rskj-core/src/main/java/co/rsk/net/sync/DownloadingHeadersSyncState.java

```diff
@@ -0,0 +1,79 @@
+package co.rsk.net.sync;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.core.BlockIdentifier;
+import org.ethereum.util.ByteUtil;
+
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+
+public class DownloadingHeadersSyncState extends BaseSyncState {
```

Comment:
We can get to this state multiple times is syncing was stopped. We should remember what headers were already downloaded.

Reply:
The sync process will restart from the beginning, being limited the amount of bodies to download makes more reasonable to discard headers if the peer is not reliable to download blocks.